core/services/job/orm.go

```diff
@@ -456,8 +456,21 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 		jb.PipelineSpecID = pipelineSpecID
 
 		err = o.InsertJob(jb, pg.WithQueryer(tx))
+		if err != nil {
+			return errors.Wrap(err, "failed to insert job")
+		}
 		jobID = jb.ID
-		return errors.Wrap(err, "failed to insert job")
+
+		// Prevents trying to insert a PRIMARY job_pipeline_specs record when one already exists
+		// If none exists, it will insert a PRIMARY record
+		var alreadyHasPrimary bool
+		err = tx.Get(&alreadyHasPrimary, `SELECT EXISTS (SELECT 1 FROM job_pipeline_specs WHERE job_id = $1 AND is_primary = true)`, jobID)
```

Comment:
@vyzaldysanchez If we are adding it via this path, I think we can assume it's a primary (isPrimary should basically act in the same way as the current foreign key).

In other words we should just fail if for some reason it already exists (I don't even think this case is possible since the job is created on line 458, unless it's doing an upsert rather than an insert).

Reply:
This is resolved. We are just always assuming that whenever we create the job via this path, the record is a primary.